### ISSUE TITLE
Vertical exaggeration for Profiles

### ISSUE THREAD

**Issue by Esteban82** [GenericMappingTools/gmt#4367]
This is a feature **suggest** in order to improve profiles. I have my commands to do it (in DOS, sorry, I will change to bash soon).

1. When I run basemap/plot the map scale is inform in the terminal. But in the case of a profile (-JX), that scale is the horizontal one and may NOT be the same as the vertical scale.

2. From the horizontal and vertical scale, a vertical exaggeration can be calculated.

Suggestions: The vertical scale and exaggeration could be shown in the terminal. 

Is it possible to save that information (horizontal and vertical scales, and vertical exaggeration) so it can be plot in the figure? I have to do the maths, show the results in the terminal and them plot it with:

```
REM	Factor for the Horizontal (FH) and Vertical (FV) to convert the graphic units (cm) to reals (m, km). 
REM	km-cm=100000, m-cm=100
	SET	FH=100000
	SET	FV=100

REM  	Save the variables to do the math
	echo %Max% > "temp_Max"
	echo %Min% > "temp_Min"
	echo %H%   > "temp_H"
	echo %KM%  > "temp_KM"
	echo %L%   > "temp_L"

REM	Show in Terminal
REM	--------------------------------------------------------------
	echo Horizontal Scale=
	gmt math "temp_KM" "temp_L" DIV %FH% MUL = 
	gmt math "temp_KM" "temp_L" DIV %FH% MUL = "temp_Esc_Hz"

	echo Vertical Scale =  
	gmt math "temp_Max" "temp_Min" SUB "temp_H" DIV %FV% MUL =
	gmt math "temp_Max" "temp_Min" SUB "temp_H" DIV %FV% MUL = "temp_Esc_Ve"

	echo Vertical Exaggeration=
	gmt math "temp_Esc_Hz" "temp_Esc_Ve" DIV =


REM	Add info in the figure
REM	-------------------------------------------------------------
REM	Data from terminal
	echo Esc. Hz. = 1:37.797.333 | gmt pstext -R -J -O -K -F+cBL+f9p -Gwhite -W1 >> %OUT%
	echo Esc. Ve. = 1:460.000    | gmt pstext -R -J -O -K -F+cBC+f9p -Gwhite -W1 >> %OUT%
	echo Ex.Vert. = 82           | gmt pstext -R -J -O -K -F+cBR+f9p -Gwhite -W1 >> %OUT%
```
![EJ11 1_Perfil_Topografico_5](https://user-images.githubusercontent.com/17908951/97066691-c08d7480-158d-11eb-9ee0-8d1315336c0d.png)

******************
Also, I thik it would be nice to have a graphical vertical exaggeration. In other words, to show how the angles are deformed:

```
REM	Angulos de Inclinacion
	echo	0	0	>  "temp_angulos"
	echo	1	0	>> "temp_angulos"
	echo	> 1\232		>> "temp_angulos"
	echo	0	0	>> "temp_angulos"
	echo	1.000	0.017	>> "temp_angulos"
	echo	> 5\232		>> "temp_angulos"
	echo	0	0	>> "temp_angulos"
	echo	0.996	0.087	>> "temp_angulos"
	echo	>10\232		>> "temp_angulos"
	echo	0	0	>> "temp_angulos"
	echo	0.985	0.174	>> "temp_angulos"
	echo	>15\232		>> "temp_angulos"
	echo	0	0	>> "temp_angulos"
	echo	0.966	0.259	>> "temp_angulos"
	echo	>20\232		>> "temp_angulos"
	gmt psxy -J -R -O -K >> %OUT% "temp_angulos" -Wthinnest

	echo	1.000	0.017	1\232	| gmt pstext -R -J -O -K >> %OUT% -Xa0.25
	echo	0.996	0.087	5\232	| gmt pstext -R -J -O -K >> %OUT% -Xa0.25 -Ya-0.02
	echo	0.985	0.174	10\232	| gmt pstext -R -J -O -K >> %OUT% -Xa0.25 -Ya-0.04
	echo	0.966	0.259	15\232	| gmt pstext -R -J -O -K >> %OUT% -Xa0.24 -Ya-0.06
	echo	0.940	0.342	20\232	| gmt pstext -R -J -O -K >> %OUT% -Xa0.23 -Ya-0.09
```

Batch script for the angles figures (with extension change to txt): [Angles_Vertical_Exaggeration.txt](https://github.com/GenericMappingTools/gmt/files/5432366/Angles_Vertical_Exaggeration.txt)

![4 3_MapaVectores](https://user-images.githubusercontent.com/17908951/97066748-4c9f9c00-158e-11eb-9e68-779c19196ca5.png)

![4 3_MapaVectores_15](https://user-images.githubusercontent.com/17908951/97066740-3eea1680-158e-11eb-91f2-553ef04274a4.png)

**Comment by PaulWessel** [comment]
Hi @Esteban82.  Seems there are two suggestions

1. Compute the vertical exaggeration for -JX|x projections and report it.
2. Offer a simple scheme of symbol to indicate angles.

I think we certainly could do the first, and report it in the INFORMATION for now.  See the PR above.  For now we have no way to know if these are vertical exaggerations or not.
As for the special symbol: Not sure what is the best way to handle this.  Not sure if a custom symbol .def file could be made.

**Comment by Esteban82** [comment]
Ok. thanks for 1). 

About 2), yes, I see that a cartesian graph it is not necessarily a profile, and so it could be confusing. It was just an idea. Thanks.

**Comment by Esteban82** [comment]
I' ve been thinking. Maybe for the case 2) a new module can be created (profile?) for the particular cases of profiles. It would be based on basemap with -JX. It would be something like grdfff and gravfft. Or maybe pscoupe could be modified/improve??

**Comment by PaulWessel** [comment]
Actually, I am fleshing out a new scale option for the angle exaggerations for basemap, e.g., something like

**-E**_repoint_**+w**_width_[/_height_]**+a**_angles_**+p**_pen_

where _refpoint_ is similar to all the other map embellishment placements, _width_ is the width of the angular scale (with optional _height_ - otherwise auto], _angles_ is a list of angles to draw (e.g., 1,2,5,10,30), and _pen_ is for the line.  If _height_ is set, then some lines may have annotations where they exit the bottom bonding box, otherwise annotations will be at the right end.

I don't think a **profile** module is needed, but this can be discussed further.  It is not a bad idea.

**Comment by Esteban82** [comment]
Ok, nice!!

A particular feature of the **profile** module could be to make real profiles (taking account the earth curvature) like the figure below:
![earth-discontinuities2](https://user-images.githubusercontent.com/17908951/98497928-3f3c1000-2224-11eb-9f6e-6bda468fea8b.png)

It could be helpfull for regional and deep profiles used in seismology. For example a calculated that for this profile of the [unavco lessons](https://github.com/GenericMappingTools/2020-unavco-course/blob/master/seismology/japan-beachballs-sideview.png) the distance at the bottom of the profile (at 650 km) should be ca. 1500 km (and not 1700 km, i.e. 10% smaller). I know that this could be done with a polar projection but I think it is a bit more difficult than a rectangular profile and maybe this could be done under the hood by this new module.

**Comment by PaulWessel** [comment]
Yes, I think this could be useful.  Maybe @seisman and @joa-quim can add some thoughts before I flesh out what profile should be.

**Comment by joa-quim** [comment]
We have talked about replicating the Julia *plot* avatars: *lines, scatter, bar, bar3, arrows* that would be just subsets of **psxy|z** with simpler documentation. I think this *profile* would go into *lines*. *Lines* is a common name in other plotting packages but *profile* is not.

**Comment by PaulWessel** [comment]
Yes, that is still possible - although I think **lines** will also plot lines in an x-y or lon/lat setting.  However, I think **profile** is a different beast.  Unlike **psxy** (and possible avatars), **profile** would explicitly be a cross-section and profiler tool, with all the machinery for lon/lat or x/y mapping _unavailable_.  It opens up for better presentation of profile-specific things, like the Earth crossection above as well as options/embellishments that only makes sense in a profile, such as the vertical exaggeration indicators and labels. While we should be careful of mission creep, **profile** _could_ for instance plot a cross-section of a 3-D grid.  It could, like @Esteban82 suggest, learn a few tricks from **pscoupe** to establish the profile given two points on the map rather than the regular **-**R_xmin/xmax/ymin/ymax_ stuff.  Currently, setting up a **psxy** profile plot between two location points requires some guru-level planning via **project**, not to mention plotting grids that way, especially 3-D grids.  We now have the ability to call **grdinterpolate** from such a profile module to make this simple.
Anyway, just giving more ideas here - happy to here counter-points on this.  Clearly, a **profile** module will not be ready for 6.2 anyway.

**Comment by PaulWessel** [comment]
Note that such a module would simply build a suitable **psxy** command under the hood, so it is not that much work to implement once we had a clear agenda on it.

**Comment by PaulWessel** [comment]
Given that we want to avoid modules that expect different types of input files, perhaps a future **crossection** module is better for grids.  So **profile** would be limited to what **psxy** can plot.
